packages/query-core/src/infiniteQueryBehavior.ts

```diff
@@ -134,6 +134,10 @@ export function infiniteQueryBehavior<
           // Fetch remaining pages
           for (let i = 1; i < oldPages.length; i++) {
             promise = promise.then((pages) => {
+              context.dispatch({
+                type: 'pageFetched',
+              })
```

Comment:
shouldn't we dispatch this event at the end of the `fetchPage` function? That way, it would be in one place, right after really fetching a page ...